### PR TITLE
This PR implements the groups and identities endpoints

### LINKED ISSUE #344: Support user.openshift.io/Identity API
The `Identity` and `UserIdentityMapping` API are currently not supported by openshift-restclient-java. It would be great if we can add them to increase the API coverage. 

See: https://docs.okd.io/latest/rest_api/apis-user.openshift.io/v1.Identity.html

### PR DESCRIPTION
**Pull request by klenkes74**
This PR should resolve #344. Since I could not get a hold on the UserIdentityMapping I did not implement that endpoint.

I also added the reference list to the already existing users endpoint.

### CODE REVIEW

File: src/main/java/com/openshift/internal/restclient/model/user/OpenShiftGroup.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    private static final String USERS = "users";
+
+    public OpenShiftGroup(ModelNode node, IClient client, Map<String, String[]> propertyKeys) {
+        super(node, client, propertyKeys);
+    }
+
+    @Override
+    public String getUID() {
+        return asString("metadata.uid");
```
Comment (adietish) [inline src/main/java/com/openshift/internal/restclient/model/user/OpenShiftGroup.java]: you might consider adding this identifier to ```com.openshift.internal.restclient.model.properties.ResourcePropertyKeys```

File: src/main/java/com/openshift/internal/restclient/model/user/OpenShiftIdentity.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @Override
+    public String getUserName() {
+        return asString(PROVIDER_USER_NAME);
+    }
+
+    @Override
+    public String getUID() {
+        return asString("metadata.uid");
```
Comment (adietish) [inline src/main/java/com/openshift/internal/restclient/model/user/OpenShiftIdentity.java]: you might consider adding this identifier to ```com.openshift.internal.restclient.model.properties.ResourcePropertyKeys```

File: src/main/java/com/openshift/internal/restclient/model/user/OpenShiftUser.java [hunk 3]
```diff
@@ -37,4 +39,13 @@ public String getUID() {
         return asString("metadata.uid");
```
Comment (adietish) [inline src/main/java/com/openshift/internal/restclient/model/user/OpenShiftUser.java]: you might consider adding this identifier to ```com.openshift.internal.restclient.model.properties.ResourcePropertyKeys```

File: src/main/java/com/openshift/internal/restclient/model/user/OpenShiftGroup.java (hunk 1 above)
Comment (klenkes74) [reply src/main/java/com/openshift/internal/restclient/model/user/OpenShiftGroup.java]: Good Idea. Fixed.

File: src/main/java/com/openshift/internal/restclient/model/user/OpenShiftIdentity.java (hunk 2 above)
Comment (klenkes74) [reply src/main/java/com/openshift/internal/restclient/model/user/OpenShiftIdentity.java]: Good Idea. Fixed.

File: src/main/java/com/openshift/internal/restclient/model/user/OpenShiftUser.java (hunk 3 above)
Comment (klenkes74) [reply src/main/java/com/openshift/internal/restclient/model/user/OpenShiftUser.java]: Good Idea. Fixed.

**Pull request merged**